service/http/handlers.go

```diff
@@ -144,3 +146,82 @@ func HandleHealthReady() http.HandlerFunc {
 		rw.WriteHeader(http.StatusOK)
 	}
 }
+
+/*
+HandleCreatePacks allows the creation of packs with pre-existing content.
+It assumes the existence of a long-running distribution (ie: MINTING status).
+It requires either the nft_flow_ids _or_ commitmentHash to be sent as part of the request. If the nft_flow_ids are
+sent, this endpoint will calculate the commitment hash
+*/
+func HandleCreatePacks(logger *log.Logger, a *app.App) http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		// Check body is not empty
+		if err := checkNonEmptyBody(r); err != nil {
+			handleError(rw, logger, err)
+			return
+		}
+
+		var reqCreatePack ReqCreatePack
+
+		// Decode JSON
+		if err := json.NewDecoder(r.Body).Decode(&reqCreatePack); err != nil {
+			handleError(rw, logger, err)
+			return
+		}
+
+		if err := reqCreatePack.Validate(); err != nil {
+			handleError(rw, logger, err)
+			return
+		}
+
+		distributionState, err := a.GetDistributionState(r.Context(), reqCreatePack.DistributionID)
+		if err != nil {
+			handleError(rw, logger, fmt.Errorf("error retrieving distribution with id %s :%w", reqCreatePack.DistributionID, err))
+			return
+		}
+
+		if distributionState != common.DistributionStateMinting {
+			handleError(rw, logger, fmt.Errorf("distribution is in invalid state with id %s :%w", distributionState, err))
+			return
+		}
+
+		addressLocation := app.AddressLocation{
+			Name:    reqCreatePack.PackReference.Name,
+			Address: reqCreatePack.PackReference.Address,
+		}
+
+		var collectibles []app.Collectible
+
+		for _, flowID := range reqCreatePack.NFTFlowIDs {
+			collectibles = append(collectibles, app.Collectible{
+				ContractReference: addressLocation,
+				FlowID:            flowID,
+			})
+		}
+
+		pack := app.Pack{
+			DistributionID:    reqCreatePack.DistributionID,
+			ContractReference: addressLocation,
+			State:             common.PackStateInit,
+			Collectibles:      collectibles,
+		}
+
+		if reqCreatePack.CommitmentHash == nil {
+			err = pack.SetCommitmentHash()
+			if err != nil {
+				handleError(rw, logger, fmt.Errorf("error setting commitment hash: %w", err))
+				return
+			}
+		} else {
+			pack.CommitmentHash = []byte(*reqCreatePack.CommitmentHash)
+		}
```

Comment:
Minor-
Would it make sense to encapsulate the logic here into `SetCommitmentHash`?
or maybe not, since we'd change SetCommitmentHash's function signature